Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -113,15 +113,22 @@ define([
      *                  be treated as if their alpha is 1.0 everywhere.  When this property is false, memory usage
      *                  and texture upload time are potentially reduced.
      * @param {GetFeatureInfoFormat[]} [options.getFeatureInfoFormats] The formats in which to get feature information at a
-     *                                 specific location when {@see UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
+     *                                 specific location when {@link UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
      *                                 parameter is not specified, feature picking is disabled.
-     *
+     * @param {Boolean} [options.deferReadiness=false] True to defer full construction of this instance until a later call to {@link UrlTemplateImageryProvider#reinitialize}.
```

Comment:
@pjcozzi @mramato I understand how to refactor `UrlTemplateImageryProvider` to allow `options` to be a promise to options, but I'm not entirely sure how to change `createTileMapServiceImageryProvider.js` to create a promise to options and pass that promise into the `UrlTemplateImageryProvider` constructor. 

On [line 169] (https://github.com/tiffanylu/cesium/blob/tileMapServiceImageryProvider/Source/Scene/createTileMapServiceImageryProvider.js#L169) in `createTileMapServiceImageryProvider`, there's a reference to the `readyPromise` and `errorEvent` properties of the returned `UrlTemplateImageryProvider` object, in the case that the `loadXML` promise fails to return a supported `tilingScheme`. 

If there is no existing instance of `UrlTemplateImageryProvider` at that point in time (i.e., no ability to create a deferred instance using `UrlTemplateImageryProvider.reinitialize`), this would not be possible. As I understand it, @kring 's rationale for creating a deferred instance was so that we could expose the `errorEvent` (and now `readyPromise`) properties of the returned imagery provider without actually needing to return said `UrlTemplateImageryProvider` instance. 

Does that make sense, or am I approaching this the wrong way?